### ISSUE TITLE
sklearn prediction output predict_proba, wine dataset, float64

### ISSUE THREAD

**Issue by hdinh** [dmlc/treelite#250]
Related to: https://github.com/dmlc/treelite/issues/153
But with another dataset (wine in this case)
Looks like no issues when X is float32.

```
from sklearn.datasets import load_wine
from sklearn.ensemble import RandomForestClassifier
import treelite
import treelite_runtime
import numpy as np

toolchain = 'gcc'
libpath = 'iris.so'
num_trees = 10
 
X, y = load_wine(return_X_y=True)
#X = X.astype(np.float32)  # works if float32
clf = RandomForestClassifier(random_state=0, n_estimators=num_trees)
clf.fit(X, y)
model = treelite.sklearn.import_model(clf)
model.export_lib(toolchain=toolchain, libpath=libpath, params={'parallel_comp':32}, verbose=True)
predictor = treelite_runtime.Predictor(libpath=libpath, verbose=True)
dmat = treelite_runtime.DMatrix(X)
out_prob = predictor.predict(dmat)

np.testing.assert_allclose(clf.predict_proba(X), out_prob)
```

**Comment by hcho3** [comment]
This behavior is expected. From https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier.fit:

> The training input samples. Internally, its dtype will be converted to dtype=np.float32.

Please convert data into float32 to ensure accuracy.
